kpagacz/wiki-on-fire
# add: added an endpoint to refresh the token

* updated the eslint configuration to avoid some false positive linting
    errors
  * improved styling in other files
  * refactored some JWT settings into the config file to avoid code
    duplication

Closes #141

Test:
* run the frontend application
* log in from the browser
* add an axios call to the /api/auth/refreshToken endpoint
* check if the token has been refreshed and the response contains the user data

If you want to check the service separately - I have user something like this:
```js
import refreshToken from "./services/refreshTokenService.js";
import config from "./config/config.example.js";
import pkg from "jsonwebtoken";
const { sign } = pkg;

const token = sign(
  {username: "test_username1", accountType: 1, accountStatus: 1},
  config.secretKey,
  {
    algorithm: config.tokenAlgorithm,
    expiresIn: config.tokenExpiration,
    issuer: "WikiOnFire"
  },
);

const res = refreshToken(token);
res.then((obj) => { console.dir(obj); });
```

Spotlight is updated. I am working on possibly passing cookies via spotlight, so fingers crossed we can test using spotlight. We'll see.
